samples/graphene-hello/preload/README.md

```diff
@@ -1,49 +1,42 @@
 
-Then, before you can run the sample, make sure you got the prerequisites for ECDSA remote attestation installed on your system. You can collectively install them with the following command:
+Then, before you can run the example, make sure you got the prerequisites for ECDSA remote attestation installed on your system. You can collectively install them with the following command:
 ```sh
 sudo apt install libsgx-quote-ex-dev
 ```
 
 ## Build
-You can build the sample as follows:
+You can build the example as follows:
 ```sh
 export GRAPHENEDIR=[PATH To Your Graphene Folder]
 make
 ```
 Then get `mr_enclave` from the build output and set it as `UniqueID` in `manifest.json`.
 
-Note that compared to the `spawn` method for this sample, the argv arguments are defined during the build and cannot be changed afterwards without rebuilding your Graphene application. To adjust the provisioned argv arguments, you can change the entry `hello.argv` in the Makefile.
+Note that compared to the `spawn` method for this example, the argv arguments are defined during the build and cannot be changed afterwards without rebuilding your Graphene application. To adjust the provisioned argv arguments, you can change the entry `hello.argv` in the Makefile.
 
 ## Run
-We assume that the Coordinator is run with the following environment variables:
-
-- EDG_COORDINATOR_MESH_ADDR=localhost:2001
-- EDG_COORDINATOR_CLIENT_ADDR=localhost:4433
-- EDG_COORDINATOR_DNS_NAMES=localhost
-- EDG_COORDINATOR_SEAL_DIR=$PWD
+Next, use the `erthost` command to start the Coordinator in a local enclave:
+```sh
+erthost ../../../build/coordinator-enclave.signed
+```
 
-Once the [Coordinator instance is running](../../BUILD.md#run-the-coordinator), upload the manifest to the Coordinator:
+The Coordinator exposes two APIs, a client API to instruct the Coordinator (port 4433) and a mesh API to communicate with your Marble (port 2001).
 
+Once the Coordinator instance is running, you can upload the mainfest to the Coordinators client API:
 ```
 curl -k --data-binary @manifest.json https://localhost:4433/manifest
 ```
 
-Now we can run our application:
+To run the application, you need to set some environment variables. The Marbles type is defined in the `manifest.json`. In this example, the manifest defines a single Marble, which is called "hello". The Marbles DNS name and the Coordinators address are used to establish a connection between the Coordinators mesh API and the Marble, and the UUID file stores a unique ID that enables a restart of the application.
 
 ```sh
-EDG_MARBLE_COORDINATOR_ADDR=localhost:2001 EDG_MARBLE_TYPE=hello EDG_MARBLE_UUID_FILE=uuid EDG_MARBLE_DNS_NAMES=localhost make run
-```
-
-## Troubleshooting
-If you receive the following error message on launch:
-
+EDG_MARBLE_TYPE=hello \
+EDG_MARBLE_COORDINATOR_ADDR=localhost:2001 \
+EDG_MARBLE_UUID_FILE=uuid \
+EDG_MARBLE_DNS_NAMES=localhost \
```

Comment:
Haven't tested it, but can we define `loader.insecure__use_host_env = 1`, yet still define additional environment variables with `loader.env.[ENVIRON] = "[VALUE]"`?

If yes, I think we should hide those in the Graphene manifest.

Reply:
I'm not familiar with Graphene, so it would be nice if you could test this and submit a suggestion.

Reply:
I'd keep them in the user's environment. The `"Don't use this in production!"` message is not ideal, but moving them into the Graphene manifest essentially makes them constants defeating the env-var purpose entirely.